CMakeLists.txt

```diff
@@ -746,7 +771,7 @@ if(EXISTS ${APPLICATION_SOURCE_DIR}/VERSION)
       -DAPP_VERSION_CUSTOMIZATION="$<TARGET_PROPERTY:app_version_h,APP_VERSION_CUSTOMIZATION>"
       ${build_version_argument}
       -P ${ZEPHYR_BASE}/cmake/gen_version_h.cmake
-    DEPENDS ${APPLICATION_SOURCE_DIR}/VERSION ${git_dependency}
+    DEPENDS ${APPLICATION_SOURCE_DIR}/VERSION ${git_dependency} ${application_git_file}
```

Comment:
not fond of Zephyr's CMakeLists.txt having to do git in repos outside of Zephyr itselft, like it's proposed in this PR.

Would using a generator expression together with the `app_version_h` target be a viable option ?
For example like this:
```
DEPENDS ${APPLICATION_SOURCE_DIR}/VERSION ${git_dependency} 
        $<TARGET_PROPERTY:app_version_h,APP_VERSION_DEPENDS>
```

then the application repo could do:
```
set_property(TARGET app_version_h PROPERTY APP_VERSION_DEPENDS <dependency>)
```

and then it would be for the external repo / build system to decide if such dependency should be a git dependency or something else.